src/modules/navigation/components/PersonalInfoBlade/PersonalInfoBlade.tsx

```diff
@@ -56,6 +73,17 @@ const PersonalInfoBlade: FC<PersonalInfoBladeBladeProps> = ({
 		}
 	};
 
+	const renderCheckbox = (renderCheckbox: boolean) => {
+		return !renderCheckbox ? (
+			<Checkbox
+				className={styles['c-personal-info-blade__checkbox']}
+				checked={newsLetterChecked}
+				label={tHtml('schrijf je in voor de nieuwsbrief')}
+				onClick={() => setNewsLetterChecked((prevState) => !prevState)}
+			/>
+		) : null;
+	};
```

Comment:
having a function and a variable be the same name is a bit confusing
also this function has access to the state variables, so you can just make all this logic internal:
```suggestion
	const renderCheckbox = () => {
		if (!shouldRenderNewsletterCheckbox) {
			return null;
		}
		return (
			<Checkbox
				className={styles['c-personal-info-blade__checkbox']}
				checked={newsLetterChecked}
				label={tHtml('schrijf je in voor de nieuwsbrief')}
				onClick={() => setNewsLetterChecked((prevState) => !prevState)}
			/>
		);
	};
```

Reply:
+ avoid render functions in general 😅
Their presence usually indicates you need a "domain component" to map your data structure to dumber ui components.
<img width="682" alt="Screenshot 2024-06-18 at 11 49 04" src="https://github.com/viaacode/hetarchief-client/assets/90180417/3e9af9d0-1518-4e95-9a5f-883257b5d433">